sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -485,6 +485,10 @@ void CurlSession::ReadStatusLineAndHeadersFromRawResponse(bool reUseInternalBUff
       // Try to fill internal buffer from socket.
       // If response is smaller than buffer, we will get back the size of the response
       bufferSize = ReadFromSocket(this->m_readBuffer, Details::c_DefaultLibcurlReaderSize);
+      if (bufferSize == 0) {
```

Comment:
Add a test case for this scenario

Reply:
Barely impossible.  Sadly, but, there's not a simple way for doing this.
We are talking about a response server that is empty, which is basically against the HTTP standard. No server would return that.
Maybe with the fake response injection,,,  But we don't have such capabilities right now